Update examples and references to v1alpha2 field names

wallrj opened the pull request:
Fixes: https://github.com/cert-manager/website/issues/302

It'd be nice to automate this a bit more, but in the end I used a script to find all the v1alpha/beta examples and manually replaced them.
Ran each of them through `fgrep -v '...' | kubectl apply -f -  --dry-run` and looked for `unknown field "..."` errors.

hzhou97 commented on content/en/docs/concepts/project-maturity.md:
```diff
@@ -31,5 +27,5 @@ many versions as possible, it is sometimes required to lose compatibility in
 
-As of cert-manager version `v0.11`, the lowest Kubernetes version supported is
-`v1.12`.
+As of cert-manager version `v1`, the lowest Kubernetes version supported is
```
I think "As of v0.11" still holds here

wallrj replied:
Yeah, I agree. But perhaps I should just remove this paragraph, because it contradicts the earlier definition of supported kubernetes versions.

meyskens replied:
it is, we added 1.11 back thanks to filter-crd

wallrj replied:
Looks like filter-crd was merged in https://github.com/jetstack/cert-manager/commit/8152159ea764b5180ace01904b5e742d7a0bbb13 and first used in cm 1.14 so I've updated the doc accordingly.

meyskens commented on content/en/docs/concepts/project-maturity.md:
```diff
@@ -6,23 +6,19 @@ type: "docs"
 
 ## Compatibility
 
 cert-manager has a hard guarantee of compatibly with the current stable upstream
```
This is not true any longer due to OpenShift 3.11 being supported for 4 years we somehow are also bound to that. Should be made into an issue to be handled later

wallrj replied:
See https://github.com/cert-manager/website/issues/305

meyskens commented on content/en/docs/concepts/webhook.md:
```diff
@@ -25,7 +25,7 @@ main functions:
   The webhook is also responsible for implementing a conversion over versions
   in the cert-manager `CustomResources` (`cert-manager.io`). This means that
-  multiple API versions can be supported, such as `v1alpha2`, `v1alpha3`
+  multiple API versions can be supported, such as `v1`, `v1beta1`
```
I would keep alpha2 in here as that is so widely used we should keep mentioning it

meyskens commented:
Looks good a few nits then LGTM

wallrj commented:
@meyskens and @hzhou97 Thanks for your reviews.
I've addressed your comments and removed the shell script and the broken API links from the PR, so hopefully the tests should all now pass.

meyskens commented:
/approve
/lgtm

let's hope the CI passes
